yogstation/code/modules/power/nuclear_engine.dm

```diff
@@ -302,14 +303,15 @@ According to players, the average usage is 160 KW, or 160,000 watts. So that's t
 			return
 		if(!istype(FR, /obj/item/twohanded/required/FuelRod/depleted)) //Depleted rods don't deplete further, they just interrupt heat
 			FR.integrity -= 10 - reactor.Efficiency/10 // Eg, 100/10 = 10, so integrity -= 0 aka no health lost, 25 = 2.5, so 8.5 health lost, so keep that efficiency up!
-	for(var/obj/item/twohanded/required/ControlRod/CR in CRS)
-		if(CR.loc != reactor || !CR in reactor.contents)
+	for(var/T in CRS)
+		var/obj/item/twohanded/required/ControlRod/CR = T
+		if(CR.loc != reactor || (!CR in reactor.contents))
 			CRS -= CR
 		if(!istype(CR,/obj/item/twohanded/required/ControlRod))
 			return
 		if(CR.integrity <= 10) //CR is spent!
-			qdel(CR)
 			reactor.say("<span class='warning'>[CR] has worn out.<span>")
+			qdel(CR)
```

Comment:
YOURE STILL USING CR AFTER YOU QDEL IT

Reply:
if he's fast enough he'll use it before it's deleted